test/e2e/operator.go

```diff
@@ -217,6 +217,51 @@ var _ = Describe("ARO Operator - RBAC", func() {
 	})
 })
 
+var _ = Describe("ARO Operator - MachineHealthCheck", func() {
+	const (
+		mhcNamespace            = "openshift-machine-api"
+		mhcName                 = "aro-machinehealthcheck"
+		mhcRemediationAlertName = "mhc-remediation-alert"
+	)
+
+	getMachineHealthCheck := func(g Gomega, ctx context.Context) {
+		_, err := clients.MachineAPI.MachineV1beta1().MachineHealthChecks(mhcNamespace).Get(ctx, mhcName, metav1.GetOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+	}
+
+	deleteMachineHealthCheck := func(g Gomega, ctx context.Context) {
+		err := clients.MachineAPI.MachineV1beta1().MachineHealthChecks(mhcNamespace).Delete(ctx, mhcName, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	getMachineHealthCheckRemediationAlertName := func(g Gomega, ctx context.Context) {
+		_, err := clients.Monitoring.MonitoringV1().PrometheusRules(mhcNamespace).Get(ctx, mhcRemediationAlertName, metav1.GetOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+	}
+
+	deleteMachineHealthCheckRemediationAlertName := func(g Gomega, ctx context.Context) {
+		err := clients.Monitoring.MonitoringV1().PrometheusRules(mhcNamespace).Delete(ctx, mhcRemediationAlertName, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	It("must be recreated if deleted", func(ctx context.Context) {
+		By("deleting the machine health check")
+		Eventually(deleteMachineHealthCheck).WithContext(ctx).Should(Succeed())
+
+		By("waiting for the machine health check to be restored")
+		Eventually(getMachineHealthCheck, "3s").WithContext(ctx).Should(Succeed())
```

Comment:
I am not sure if this means poll every 3s or timeout in 3s. Would you please use the other form

```
Eventually(getMachineHealthCheck).WithContext(ctx).WithTimeout(3*time.Second).Should(Succeed())
```

or 

```
Eventually(getMachineHealthCheck).WithContext(ctx).WithTimeout(3*time.Second).Should(Succeed())
```

It would remove the ambiguity. Thank you.